src/session_impl.cpp

```diff
@@ -5162,11 +5154,21 @@ namespace aux {
 		// proxy, and it's the same one as we're using for the tracker
 		// just tell the tracker the socks5 port we're listening on
 		if (m_socks_listen_socket && m_socks_listen_socket->is_open())
-			return m_socks_listen_socket->local_endpoint().port();
+		{
+			error_code ec;
+			tcp::endpoint endp = m_socks_listen_socket->local_endpoint(ec);
+			if (ec)
```

Comment:
It's the same, but the method `local_endpoint()` without the parameter does not exists under `BOOST_NO_EXCEPTIONS` and fails to compile